src/shims/io.rs

```diff
@@ -0,0 +1,170 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+
+use crate::stacked_borrows::Tag;
+use crate::*;
+
+pub struct FileHandler {
+    files: HashMap<i32, File>,
```

Comment:
`files` and `flags` always have the same entries, right? If so, create a `FileHandle` struct that has the `File` field and a `flag` field and only use one `HashMap`

Reply:
Yes, they have the same entries.